timeseries/src/autogluon/timeseries/models/chronos/pipeline/utils.py

```diff
@@ -8,6 +11,44 @@
 from autogluon.timeseries.dataset.ts_dataframe import TimeSeriesDataFrame
 
 
+def left_pad_and_stack_1D(tensors: List[torch.Tensor]) -> torch.Tensor:
+    max_len = max(len(c) for c in tensors)
+    padded = []
+    for c in tensors:
+        assert isinstance(c, torch.Tensor)
+        assert c.ndim == 1
+        padding = torch.full(size=(max_len - len(c),), fill_value=torch.nan, device=c.device)
+        padded.append(torch.concat((padding, c), dim=-1))
+    return torch.stack(padded)
+
+
+def download_prefix(bucket, prefix, local_path, force: bool = False, boto3_session=None) -> None:
+    import boto3
+
+    boto3_session = boto3_session or boto3.Session()
+    s3_resource = boto3_session.resource("s3")
+    bucket = s3_resource.Bucket(bucket)
+
+    for obj in bucket.objects.filter(Prefix=prefix):
+        if obj.key.endswith("/"):
+            continue
+        dest = local_path / bucket.name / obj.key
+        if not force and dest.exists():
+            continue
+        if not dest.parent.exists():
+            dest.parent.mkdir(exist_ok=True, parents=True)
+        bucket.download_file(obj.key, str(dest))
+
+
+def cache_model_from_s3(s3_uri: str, force=False):
+    assert re.match("^s3://([^/]+)/(.*?([^/]+)/?)$", s3_uri) is not None, f"Not a valid S3 URI: {s3_uri}"
+    cache_home = Path(os.environ.get("XGD_CACHE_HOME", os.path.expanduser("~/.cache")))
```

Comment:
1. Should this be `XDG_CACHE_HOME` instead of `XGD_CACHE_HOME`?
2. Does this work on Windows (especially the second part with both `~`, `/` and `.`)?